### DISCUSSION TITLE
Export geo_model for other interfaces

### DISCUSSION THREAD

**Discussion by Changwan-planet** [gempy-project/gempy#880] [Q&A]
Hello, 

I made my geo_model as a first version with your nice help! I would like to evaluate it in the other 3D visualization software such as blender or pyvista. 
I saw other references below. 
https://www.youtube.com/watch?v=pDDMwdMjn3A
https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/69_Export_GemPy_model_into_blender.html?highlight=Blender

But, in the first step, I met an obstacle: get_surfaces.
```
gp.get_surfaces(geo_model)
```

get_surfaces is not working with this error message.
```
module 'gempy' has no attribute 'get_surfaces
```

I think things are gonna change in the gempy. 
Please help me out!

Thanks you so much.

**Comment by javoha** [comment]
Hi,
as far as I know this changed between the legacy version and gempy v3. I can get access to the meshes using:

`geo_model.solutions.dc_meshes`

Is this what you were looking for?

Cheers
Jan

**Comment by Changwan-planet** [comment]
Thank you for your answer. 

I hope to save the geo model to run in other software such as blender. 
"geo_data.solutions.dc_meshes" is working.


```
Work in progress

GemPy3 model serialization is currently being redisigned. Therefore, at the current version, there is not a build in method to save the model. However, since now the data model should be completely robust, you should be able to save the [gempy.core.data.GeoModel](https://docs.gempy.org/Modeling%20Classes/gempy.core.data.GeoModel.html#gempy.core.data.GeoModel) and all its attributes using the standard python library [pickle](https://docs.python.org/3/library/pickle.html)
```
Based on this contents,
even though I use the function "geo_data.solutions.dc_meshes", I cannot export the model in gempy v3. Right?


*Second question is about 2D-sections.
My plan was to export the geo_model for investigating the 2D section view of the model.  
At this moment I cannot do that.

Thankfully, I found [this](https://docs.gempy.org/tutorials/ch1_fundamentals/ch1_3b_cross_sections.html).
But I met another obstacle.  

```
gpv.plot_2d(geo_data, section_names=['section1'], show_data=False, legend=False)
```
This part show me the error:

```
cannot reshape array of size 0 into shape (50,50)
```

How to fix this issue?
I think the there is the problem in the resolution of "section_dict".

```
gp.set_section_grid(
    grid=geo_data.grid,
    section_dict={
        'section1': ([216185,342185], [216195,342195], [50,10])
        ,'section2': ([216185,342185], [216195,342185], [50,50])       
    }  
)
```
Please refer to my uploaded files.
[Question_20240502.zip](https://github.com/gempy-project/gempy/files/15187633/Question_20240502.zip)

*Third question.
 Points and Orientations in my data include the fault. After I compute the model, Can I make the fault transparent in the graph?

**Reply by javoha** [reply]
Hi,
let's go through a couple of your points. There is no way to save the complete model at the moment, but due to the structure you shoul be able to extract and save all the results using `.solutions` easily. Just some examples would be:
`geo_data.solutions.raw_arrays.lith_block`
`geo_data.solutions.raw_arrays.vertices`

Mixed news on the custom sections. Your error results from a flow issue. When defining custom sections you are creating custom grids that need to be computed with the model, so you need to define the sections before computing the model. Thats why in your case the result is empty and cannot be shaped into the desired resolution. Here is a small example with your data: 
[Small example.zip](https://github.com/gempy-project/gempy/files/15197876/Small.example.zip)

So while this works in general, we just encountered some issues with the location of these custom sections. We are working on it but you should be cautious for now. Hopefully this will be fixed soonish.

Final questions_ Not sure if you can set just the fault transparent. Definitely there is a way to hack that with pyvista, but I don't think we have a function for that in gempy at the moment.

**Reply by Changwan-planet** [reply]
@javoha 
I really appreciate your help. 
I will try your guides for my geo model.
